sparse_ho/criterion.py

```diff
@@ -20,50 +20,40 @@ class HeldOutMSE():
     # cv as [(train, test)] ie directly the indices of the train
     # and test splits.
 
-    def __init__(self, X_val, y_val, model, convexify=False,
-                 gamma_convex=1e-2, X_test=None, y_test=None):
+    def __init__(
+            self, idx_train, idx_val, X_test=None, y_test=None,):
         """
         Parameters
         ----------
-        X_val : {ndarray, sparse matrix} of shape (n_samples, n_features)
-            Validation data
-        y_val : {ndarray, sparse matrix} of shape (n_samples,)
             Validation target
-        model: object of the class Model (e.g. Lasso or SparseLogreg)
         X_test : {ndarray, sparse matrix} of shape (n_samples_test, n_features)
             Test data
-        convexify: bool
-            this param should be remove from here XXX
-        gamma_convex: bool
-            this param should be removed from here XXX
         y_test : {ndarray, sparse matrix} of (n_samples_test)
             Test target
         """
-        self.X_val = X_val
-        self.y_val = y_val
         self.X_test = X_test
         self.y_test = y_test
-        self.model = model
-        self.convexify = convexify
-        self.gamma_convex = gamma_convex
+        self.idx_train = idx_train
+        self.idx_val = idx_val
 
         self.mask0 = None
         self.dense0 = None
         self.quantity_to_warm_start = None
         self.val_test = None
         self.rmse = None
 
-    def get_v(self, mask, dense):
-        return 2 * (self.X_val[:, mask].T @ (
-            self.X_val[:, mask] @ dense - self.y_val)) / self.X_val.shape[0]
-
-    def value(self, mask, dense):
+    def get_mse_val(self, X, y, mask, dense):
```

Comment:
@QB3  for me get_value was better, a consistent name across all `Criterion`

Reply:
I am going for ```value_outer_crit``` in all criterions